### PR TITLE
refactor(Core/SpellMgr): Remove nested ifs for binary evaluation.

### PR DESCRIPTION
**Pull request by heyitsbench**
## Changes Proposed:

This PR proposes changes to:
-  [X] Core (units, players, creatures, game systems).

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/pull/19760#discussion_r1733342359

## Tests Performed:

This PR:
- [X] Has not been tested.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Spells/SpellMgr.cpp [hunk 1]
```diff
@@ -3065,16 +3065,28 @@ void SpellMgr::LoadSpellInfoCustomAttributes()
...
+                                continue;
+
+                            if (spellInfo->SpellFamilyName == SPELLFAMILY_WARLOCK &&
+                                ((spellInfo->SpellFamilyFlags[1] & 0x40000) || (spellInfo->SpellFamilyFlags[0] & 0x4000))) // Haunt/Drain Soul
+                                continue;
+
+                            spellInfo->AttributesCu |= SPELL_ATTR0_CU_BINARY_SPELL;
+                            break;
+
                             continue;
```
Comment (walkline) [inline src/server/game/Spells/SpellMgr.cpp:3090]: This line can be removed.

**Review by walkline — COMMENTED** [review]
Thank you!
There is tiny nitpick.

File: src/server/game/Spells/SpellMgr.cpp (hunk 1 above)
Comment (heyitsbench) [reply src/server/game/Spells/SpellMgr.cpp:3089]: Done, unless you meant the continue. 😛

Comment (walkline) [reply src/server/game/Spells/SpellMgr.cpp:3089]: Sorry, I meant `continue;` line.

Comment (heyitsbench) [reply src/server/game/Spells/SpellMgr.cpp]: Actually done now, thank you. ❤️

File: src/server/game/Spells/SpellMgr.cpp [hunk 2]
```diff
@@ -3065,17 +3065,27 @@ void SpellMgr::LoadSpellInfoCustomAttributes()
...
-                                            if (spellInfo->SpellFamilyName != SPELLFAMILY_WARLOCK || !(spellInfo->SpellFamilyFlags[1] & 0x40000)) // Haunt
-                                                if (spellInfo->SpellFamilyName != SPELLFAMILY_WARLOCK || !(spellInfo->SpellFamilyFlags[0] & 0x4000)) // Drain Soul
-                                                {
-                                                    spellInfo->AttributesCu |= SPELL_ATTR0_CU_BINARY_SPELL;
-                                                    break;
-                                                }
-                            continue;
+                            if (!spellInfo->Effects[j].CalcValue() &&
+                                !(spellInfo->Effects[j].Effect == SPELL_EFFECT_INTERRUPT_CAST || spellInfo->HasAttribute(SPELL_ATTR0_CU_DONT_BREAK_STEALTH)) ||
+                                spellInfo->HasAttribute(SPELL_ATTR0_NO_IMMUNITIES))
```
Comment (walkline) [inline src/server/game/Spells/SpellMgr.cpp]: There is warning here.
Can you try this one:
```
if (!(spellInfo->Effects[j].CalcValue() || 
      ((spellInfo->Effects[j].Effect == SPELL_EFFECT_INTERRUPT_CAST || spellInfo->HasAttribute(SPELL_ATTR0_CU_DONT_BREAK_STEALTH)) && 
       !spellInfo->HasAttribute(SPELL_ATTR0_NO_IMMUNITIES))))
```

**Review by Nyeriah — APPROVED** [review]

**Pull request merged**